units/stats_queue.go

```diff
@@ -0,0 +1,74 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/scheduler"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/logging"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const queueStatsCollectorJobName = "queue-stats-collector"
+
+func init() {
+	registry.AddJobType(queueStatsCollectorJobName,
+		func() amboy.Job { return makeQueueStatsCollector() })
+}
+
+type queueStatsCollector struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	logger   grip.Journaler
+}
+
+func NewQueueStatsCollector(id string) amboy.Job {
+	j := makeQueueStatsCollector()
+	j.SetID(fmt.Sprintf("%s-%s", queueStatsCollectorJobName, id))
+	return j
+}
+
+func makeQueueStatsCollector() *queueStatsCollector {
+	j := &queueStatsCollector{
+		logger: logging.MakeGrip(grip.GetSender()),
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    queueStatsCollectorJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func (j *queueStatsCollector) Run(_ context.Context) {
+	defer j.MarkComplete()
+
+	if j.logger == nil {
+		j.logger = logging.MakeGrip(grip.GetSender())
+	}
+
+	env := evergreen.GetEnvironment()
+	settings := env.Settings()
+	if env != nil {
+		j.AddError(errors.New("settings is nil"))
+		return
+	}
+
+	finder := scheduler.GetTaskFinder(settings.Scheduler.TaskFinder)
+	tasks, err := finder("")
```

Comment:
should be the task finder from the scheduler settings

Reply:
That's what the line above is.